docs/implementation_plans/2026-01-29_timeline_smart_folding.md

```
@@ -0,0 +1,771 @@
+  );
+
+  await tester.tap(find.byType(CompressedTimelineRegion));
+  expect(tapped, isTrue);
+});
+```
+
+---
+
+## Visual Mockup
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)